### PR TITLE
fix: panic in SendCheckpointToBTC

### PR DESCRIPTION
**Pull request by guoshijiang**
resolve this issue: https://github.com/babylonlabs-io/vigilante/issues/137

### CODE REVIEW

**Review by SebastianElvis — APPROVED** [review]
Lgtm, thanks for the fix!

Probably @gitferry can take a look as well

**Comment by gitferry** [comment]
Thanks for the report! I think the issue is valid. Basically, in the case of https://github.com/babylonlabs-io/vigilante/issues/137
1. the submitter failed to submit the first tx of a checkpoint
2. it fell into `MaybeResubmitSecondCheckpointTx` to try to re-submit the second tx of the `lastSubmittedCheckpoint`. Then the panic happened because the submitter was initiated with empty `lastSubmittedCheckpoint`
  - the check `if durSeconds < rl.config.ResendIntervalSeconds` can pass is because the submiter was initiated with empty `lastSubmittedCheckpoint` 
Therefore, I think a proper fix is to `continue` the for loop if `SendCheckpointToBTC` returns error. This ensures that an empty `lastSubmittedCheckpoint` will not go to `MaybeResubmitSecondCheckpointTx`. @Lazar955 wdyt?

**Comment by gitferry** [comment]
Thanks for the report! I think the issue is valid. Basically, in the case of https://github.com/babylonlabs-io/vigilante/issues/137
1. the submitter failed to submit the first tx of a checkpoint
2. it fell into `MaybeResubmitSecondCheckpointTx` to try to re-submit the second tx of the `lastSubmittedCheckpoint`. Then the panic happened because the submitter was initiated with empty `lastSubmittedCheckpoint`
  - the check `if durSeconds < rl.config.ResendIntervalSeconds` can pass is because the submiter was initiated with empty `lastSubmittedCheckpoint` 
Therefore, I think a proper fix is to `continue` the for loop if `SendCheckpointToBTC` returns error. This ensures that an empty `lastSubmittedCheckpoint` will not go to `MaybeResubmitSecondCheckpointTx`. @Lazar955 wdyt?

**Comment by Lazar955** [comment]
> Thanks for the report! I think the issue is valid. Basically, in the case of #137
> 
> 1. the submitter failed to submit the first tx of a checkpoint
> 2. it fell into `MaybeResubmitSecondCheckpointTx` to try to re-submit the second tx of the `lastSubmittedCheckpoint`. Then the panic happened because the submitter was initiated with empty `lastSubmittedCheckpoint`
>    
>    * the check `if durSeconds < rl.config.ResendIntervalSeconds` can pass is because the submiter was initiated with empty `lastSubmittedCheckpoint`
> 
> Therefore, I think a proper fix is to `continue` the for loop if `SendCheckpointToBTC` returns error. This ensures that an empty `lastSubmittedCheckpoint` will not go to `MaybeResubmitSecondCheckpointTx`.
> 
> @Lazar955 wdyt?

Agreed with @gitferry, err happened in `SendCheckpointToBTC` therefore we shouldn't execute `MaybeResubmitSecondCheckpointTx` but rather `continue`.

@guoshijiang can you amend your PR?

**Comment by guoshijiang** [comment]
> > Thanks for the report! I think the issue is valid. Basically, in the case of #137
> > 
> > 1. the submitter failed to submit the first tx of a checkpoint
> > 2. it fell into `MaybeResubmitSecondCheckpointTx` to try to re-submit the second tx of the `lastSubmittedCheckpoint`. Then the panic happened because the submitter was initiated with empty `lastSubmittedCheckpoint`
> >    
> >    * the check `if durSeconds < rl.config.ResendIntervalSeconds` can pass is because the submiter was initiated with empty `lastSubmittedCheckpoint`
> > 
> > Therefore, I think a proper fix is to `continue` the for loop if `SendCheckpointToBTC` returns error. This ensures that an empty `lastSubmittedCheckpoint` will not go to `MaybeResubmitSecondCheckpointTx`.
> > @Lazar955 wdyt?
> 
> Agreed with @gitferry, err happened in `SendCheckpointToBTC` therefore we shouldn't execute `MaybeResubmitSecondCheckpointTx` but rather `continue`.
> 
> @guoshijiang can you amend your PR?

update

**Comment by Lazar955** [comment]
> > > Thanks for the report! I think the issue is valid. Basically, in the case of #137
> > > 
> > > 1. the submitter failed to submit the first tx of a checkpoint
> > > 2. it fell into `MaybeResubmitSecondCheckpointTx` to try to re-submit the second tx of the `lastSubmittedCheckpoint`. Then the panic happened because the submitter was initiated with empty `lastSubmittedCheckpoint`
> > >    
> > >    * the check `if durSeconds < rl.config.ResendIntervalSeconds` can pass is because the submiter was initiated with empty `lastSubmittedCheckpoint`
> > > 
> > > Therefore, I think a proper fix is to `continue` the for loop if `SendCheckpointToBTC` returns error. This ensures that an empty `lastSubmittedCheckpoint` will not go to `MaybeResubmitSecondCheckpointTx`.
> > > @Lazar955 wdyt?
> > 
> > 
> > Agreed with @gitferry, err happened in `SendCheckpointToBTC` therefore we shouldn't execute `MaybeResubmitSecondCheckpointTx` but rather `continue`.
> > @guoshijiang can you amend your PR?
> 
> update

Will approve once CI passes. Update the changelog file as well (see the previous updates)

File: CHANGELOG.md [hunk 1]
```diff
@@ -37,6 +37,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 
 ## Unreleased
 
+### Bug Fixes
+* [#138](https://github.com/babylonlabs-io/vigilante/pull/138) fix: panic in SendCheckpointToBTC
+
```
Comment (Lazar955) [inline CHANGELOG.md:42]: rm extra new line

**Review by Lazar955 — APPROVED** [review]
lgtm, thanks for your contribution

**Pull request merged**